src/pages/_app.tsx

```diff
@@ -1,199 +1,134 @@
-        hideProgressBar: false,
-        closeOnClick: false,
-        pauseOnHover: true,
-        draggable: true,
-        progress: undefined,
-        theme: 'light',
-      })
-    }
-  },
-  onLoadingSlow(key, config) {
-    toast.warn(t`Loading seems to take a bit longer than usual. Please hold the line…`, {
-      toastId: key,
-      delay: 2000,
-      autoClose: false,
-      position: 'bottom-right',
-      hideProgressBar: false,
-      closeOnClick: false,
-      pauseOnHover: true,
-      draggable: true,
-      progress: undefined,
-      theme: 'light',
-    })
-  },
-  onSuccess(data, key, config) {
-    toast.dismiss(key)
-  },
-}
-
-export default function MyApp(props: AppProps<ExtraProps> & AppPropsWithLayout) {
-  const { Component, pageProps } = props
+export default function MyApp(
+  props: AppProps<ExtraProps> & AppPropsWithLayout,
+) {
+  const { Component, pageProps } = props;
   const {
-    userAgentString, app, session, languageTags, ipCountryCode, environmentVariables,
-  } = pageProps
-
-  // can be done always, if it's empty, it won't overwrite anything
-  addToEnvironment(environmentVariables)
-  const environment = getEnvironment()
-
-  const contexts: ContextAndValue<any>[] = [
-    [UserAgentContext, parseUserAgentString(userAgentString)],
-    [AppContext, app],
-    [HostnameContext, app.hostname],
-    [LanguageTagContext, { languageTags }],
-    [CountryContext, ipCountryCode],
-    [EnvContext, environment],
-  ]
+    userAgentString,
+    session,
+    languageTags,
+    countryCode,
+    environmentVariables,
+    hostname,
+  } = pageProps;
 
   // Use the layout defined at the page level, if available
-  const getLayout = Component.getLayout ?? ((page) => page)
+  const getLayout = Component.getLayout ?? ((page) => page);
 
   return (
     <React.StrictMode>
       <Head />
       <StyledComponentsRegistry>
         <HotkeysProvider>
           <SessionProvider session={session}>
-            <SWRConfig value={globalSWRConfig}>
-              {composeContexts(
-                contexts,
-                getLayout(<Component {...pageProps} />),
-              )}
-            </SWRConfig>
+            <SWRConfigProvider>
+              <EnvContextProvider environmentVariables={environmentVariables}>
+                <HostnameContextProvider hostname={hostname}>
+                  <UserAgentContextProvider userAgentString={userAgentString}>
+                    <CountryContext.Provider value={countryCode}>
+                      <LanguageCodeContextProvider languageTags={languageTags}>
+                        <AppContextProvider>
+                          {getLayout(<Component />)}
+                        </AppContextProvider>
+                      </LanguageCodeContextProvider>
+                    </CountryContext.Provider>
+                  </UserAgentContextProvider>
+                </HostnameContextProvider>
+              </EnvContextProvider>
+            </SWRConfigProvider>
```

Comment:
While I personally liked the previous nesting function, this is easier to understand (and adds æsthetics of an arrow pointing forward), so I still like it 🤝

Reply:
➡️